### PR TITLE
Adds admin logs anytime a non observing admin ghosts or views the player panel

### PR DESCRIPTION
**Pull request by Sarcalogo**
Admins can ghost and see a shit ton of round information and it's not even logged. It's very easy to get away with. In the same vein, an Admin can open the player panel, hover over someone's name and see if they are a traitor or not, absolutely no way to get caught. This fixes it, so if a non observing admin ghosts, opens the player panel, or shows the player panel of a specific player, other admins are notified.  That's all.

### CODE REVIEW

**Comment by Pyko1** [comment]
I see spam, at least make it so that it does not do that for observering admins.

**Comment by Gandalf2k15** [comment]
We already have enough spam, I don't want to have more... You have very little faith in the admins clearly.

**Comment by Nebyab** [comment]
Unlike everyone else apparently, I do believe this has some merit, though is it possible for you to feed it into the admin-logs that don't show up in asay as that could get incredibly spammy. Its a valuable tool for double-checking these sorts of incidents, but ideally we can do that without compromising the actual ability of admins to administrate.

**Comment by Sarcalogo** [comment]
It's done Pyko.

**Comment by Sarcalogo** [comment]
I have absolutely no faith at all actually. If someone like Agent Speer, who literally only plays to collect valids, who suicides anytime he does not get antag, and the times he doesn't plays as power gaming  Security  who kills everyone, why would he not? They have literally 0 chance of being caught, why would they not? Because some 16 year old kid who only plays the game for valids has some sort of moral objection against it? Nonsense. This isn't bringing up recent drama either. You can say "muh spam" all you want, but how often do non observing admins need to admin ghost anyway? 

Now it's pretty obvious what's going on if an admin aghosts for no reason (no admin helps). Right now there is literally nothing stopping you from aghosting as the Head of Security and jumping over to the guy they know is a traitor, to find out where he has been hiding the whole round. It would take 10 seconds, and if asked "how'd you know where he was" or "how'd you know who he was" you could just say, oh I found him, or oh I random searched him.

**Comment by Pyko1** [comment]
I see no harm in this as long as it does not cause excessive spam.
And having these alerts restricted only on playing admins should take care of it.

Even tho, if someone wishes to abuse he will just most likely use other methods to do so.

**Comment by trifornt-tinfornt** [comment]
Hornygranny PR.

Besides that, eh. Not sure if it would actually help admins. We had Hackole on the team and he didn't aghost or use the player panel to cheat - and other admins would catch on if he did.

**Comment by Pyko1** [comment]
Speaking of other methods, maybe disable ghost tab for aghosts.
Since its silly how follow alerts but ghost orbit does not.

But ultimately the rotten apples well be caught sooner or later.

**Comment by Nebyab** [comment]
>Hornygranny PR.

I'm pretty sure this is by definition the opposite of an HG pr, as its trying to stop, not encourage admin abuse.

**Comment by nalarmg** [comment]
This won't really change or help much - admins when playing regularly get the same, if not more abusable information from adminhelps. 

I can understand the intent, but this seems like it'll just be more information spam for the administrators to deal with more than much else.

**Comment by Carbonhell** [comment]
Change the message_admins to log_admin and we're good.

**Comment by Sarcalogo** [comment]
>Other admins would catch on if he did

Pretty impossible to do considering it's not even logged.

1. Open the player panel.
2. Hover your mouse over all the players, stop when you see big red "Antagonist"
3. "Randomly" search this player along with a few others
4. If no antag gear, "randomly" search them again.

If asked why you did it, say :p I'm Hackole I randomly searched him! Come on dude, it's easy as fuck to abuse. Admin ghosting you have a tiny ass chance to get caught if someone is orbiting you or sees your ghost in the 1 second it takes to jump to whoever you want to track.

**Comment by Pyko1** [comment]
yeah log_admins would be perfect

**Comment by Sarcalogo** [comment]
>Change the message_admins to log_admin and we're good.
But then it's not immediately noticeable, it's not like anyone checks the logs. Eh... I dunno, maybe.. I guess it's better than nothing but..

**Comment by Sarcalogo** [comment]
Eh... I guess I could change it, I guess it's at least possible to catch people doing it now if someone is vigilant enough.

**Comment by Carbonhell** [comment]
Avoid baseless claims. We do indeed check the logs. Thank you for the PR anyway.

File: code/modules/admin/admin_verbs.dm [hunk 1]
```diff
@@ -370,8 +374,6 @@ var/list/admin_verbs_hideable = list(
 	if(holder)
 		holder.check_antagonists()
 		log_admin("[key_name(usr)] checked antagonists.")	//for tsar~
-		if(!isobserver(usr))
```
Comment (Carbonhell) [inline code/modules/admin/admin_verbs.dm:373]: Why did you remove this?

**Comment by Sarcalogo** [comment]
Oh kek my bad. Ctrl Fing gone wrong.

**Comment by trifornt-tinfornt** [comment]
> I'm pretty sure this is by definition the opposite of an HG pr, as its trying to stop, not encourage admin abuse.
Did I actually get that backwards there?
Damn I'm dumb.

**Comment by Carbonhell** [comment]
Code's ok

**Comment by McBawbaggings** [comment]
Pretty sure all this stuff would get logged under hrefs though? +1 one though, admins should have nothing to fear if they have nothing to hide over it.

**Comment by JohnGinnane** [comment]
Code looks good!

**Comment by Carbonhell** [comment]
No baw,this won't go in hrefs

**Comment by McBawbaggings** [comment]
Isn't verb clicking href stuff already? Have I deceived myself???

**Comment by dreamcrab** [comment]
works for me, id almost like something to notify us if powergamers are checking their pda at roundstart and not doing anything. like an anti nuke-op meta check?? who knows im stupid

**Comment by doobiedoo23** [comment]
yeah but if we merge this you'll get people like speaker going HURR WHAT R U DOING BUDDY

**Comment by Iamgoofball** [comment]
@dreamcrab doesn't work anymore, patched it ages ago

:+1: of course the administration is going to whine when you take away their metagame tools

**Comment by blobbernaut** [comment]
But then it's not immediately noticeable, it's not like anyone checks the logs. Eh... I dunno, maybe.. I guess it's better than nothing but..

^Not actually true but whatever

**Comment by Iamgoofball** [comment]
well it looks like they're whining right now so i think I'm right

re: the PDA thing

unless you guys re-added Player Ready counts then there's no chance of PDA metagaming

**Comment by Voratius** [comment]
god dammit, looks like i'm gonna have to metagame in teamspeak harder...

**Comment by doobiedoo23** [comment]
awesome

**Comment by McBawbaggings** [comment]
I'm right in thinking that log_admin only writes to the .log files AND that message_admin only writes to the admins INGAME.

If so, since this uses log_admin then there should be no spam for admins but the evidence is there if an investigation is required.

**Comment by Iamgoofball** [comment]
@McBawbaggings not good enough, by that point the metagame has occured already and its too late to stop it

**Comment by McBawbaggings** [comment]
Speaking of which, if you want to go down this route, the admin version of WHO lists antags as well. Might want to throw an isobserver check on that too.

**Comment by Sarcalogo** [comment]
Nah, think I'd just remove the antagonists showing on who. Forgot about that one to be honest.

**Comment by McBawbaggings** [comment]
That's fine too.

**Comment by Memendia** [comment]
:+1:

**Comment by Sarcalogo** [comment]
So uh it's been approved by 2 heads.

**Comment by McBawbaggings** [comment]
"Nah, think I'd just remove the antagonists showing on who. Forgot about that one to be honest."
Is that going to get done though?

**Comment by Gandalf2k15** [comment]
Looking forward to people going "WAT U DING WHY U ME TAGS ME OMG"

**Comment by Sarcalogo** [comment]
Oh that's right. Silly me. I'll fix that one in a jiffy.

**Comment by McBawbaggings** [comment]
Someone please double check that

**Comment by TheOneAndOnlyCreeperJoe** [comment]
Don't disable antags from appearing entirely on the who list, only for non-observing admins. I use it all the time when I'm observing, would hate to see that go.

Other than that I am +1-ing this, despite there being like, 4 ways around this to still check if someone's an antag with badmin powers.

**Comment by Sarcalogo** [comment]
Can't think of the other 4 ways. Tell me what they are and I'll try to fix it. I realize it's been open a long time, but I don't mind if it takes a while to fix everything.

**Comment by McBawbaggings** [comment]
View Vars > Mind Datum > Special Role
I can't think 3 other ways.

**Comment by TheOneAndOnlyCreeperJoe** [comment]
1, you can enter VV and look at the mind datum. Special roles are listed there. Certain vars also change on antag status.
2, mentioning their name in asay/ahelp allows you to view what they're up to. especially helps with nerds who use names such as 'and'.
3, unreliable but check contents lets you see everything they're carrying, including traitor stuff. Can also be done trough vv.
4, unreliable but investigating hrefs let's you see some antag actions.

You also gotta imagine, events tell the admins whose made an antag, there's always ahelps coming in from both the victims and sometimes the antags themselves, in the end this only answers two of the problems.

Coming to think of it the fact that it gets logged in chat instead of only admin_log is really dumb and will just generate unnecessary clutter, because nobody really cares about admins aghosting or the fact that they're checking player panels, nobody's going to look at another admin just because they did either of the two. Logging it however, is extremely useful. That and restricting who antagonists from observing admins is dumb, because nobody benefits from it.

On hindsight I'm actually giving you a -1 until it doesn't log in chat and re-add the who antagonist check.

**Comment by Sarcalogo** [comment]
Don't think I can fix most of that. Probably only number 1. Reading the whole discussion shows that it's not logged in chat, it's admin logged.

**Comment by TheOneAndOnlyCreeperJoe** [comment]
Even then I don't want you dodging the point on not having antagonists displayed in who for observers.

**Comment by Sarcalogo** [comment]
I'll see what I can do.

**Comment by doobiedoo23** [comment]
tbh i'm certain that most of those verbs show up in hrefs creeper

your asay example is flimsy as that also leaves logs

if someone is really metagaming through check contents that'd be extremely funny and sad (also maybe logged)

**Comment by McBawbaggings** [comment]
You can't fix number 1 I don't think
Also, this isn't message_admins, it's admin_logs. This shouldn't fill the chat with needless actions.

**Comment by danhibiki** [comment]
Zero effect, just clutter if you don't catch all the other instances as well. For instance, a quick who will tell you who is antagonist and if a player is dead or alive and that isn't logged

File: code/game/verbs/who.dm [hunk 2]
```diff
@@ -68,9 +68,6 @@
 						entry += " - <font color='black'><b>DEAD</b></font>"
 				else
 					entry += " - <font color='black'><b>DEAD</b></font>"
-		if(is_special_character(C.mob))
-			entry += " - <b><font color='red'>Antagonist</font></b>"
-	entry += " (<A HREF='?_src_=holder;adminmoreinfo=\ref[C.mob]'>?</A>)"
```
Comment (Carbonhell) [inline code/game/verbs/who.dm:73]: why did you remove the ? href too, and btw i don't think this really needs to be removed either

**Comment by Carbonhell** [comment]
log_admin doesn't show up ingame. It only shows up in the logs. Whereas message_admins shows up both ingame and in logs.

**Comment by Carbonhell** [comment]
So yeah, readd the who antagonist thing and this should be good to go

**Comment by Sarcalogo** [comment]
Will finish this tomorrow. 

Must say, the who feature being in in my opinion is dumb and easily abusable and offers not much in the way of admins doing their job effectively. They can just press check antags and get the same information. Not sure why Creeper Joe wants to keep it in. Maybe he abuses it ;)?

**Comment by TheOneAndOnlyCreeperJoe** [comment]
Aaaaaaaaa

That is because who shows literaly every antag including ones that are not on check antags. Animals, zombies, drones, swarmers, Slaughter Demons, Xenos and a bunch more. I also use it to determine if the round needs to be ended. All antags dead + too many people dead for muligan? Better call it. Can't do that if it's gone. If there are xenos around I want to quickly assess how many there are without having to read every goddamn name in the game.

Sneaking in a feature removal after getting enough head PRs is cancer. The idea of removing a feature entirely because you don't trust the admins is dumb.

**Comment by Iamgoofball** [comment]
@TheOneAndOnlyCreeperJoe 
too pacific ocean for me

**Comment by Gandalf2k15** [comment]
Holy shit stfu

**Comment by Iamgoofball** [comment]
@Gandalf2k15 
all that salt and no fries?

**Comment by Sarcalogo** [comment]
Okay. Here goes. I guess I will have to create another PR later that buffs check antagonists so this easily abused feature can be removed.

**Comment by Nebyab** [comment]
@Iamgoofball please keep your shitposts to your own prs/own codebase thanks m8

**Comment by Iamgoofball** [comment]
@Nebyab 
insert salt pun here

**Comment by Sarcalogo** [comment]
This is ready and approved.

**Comment by Kokojos** [comment]
Here goes nothing
